### PR TITLE
add e2e test for edge autonomy

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <guoyao17@huawei.com>

**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Edge Autonomy: Ensure edge nodes run autonomously and the applications in edge run normally, when the cloud-edge network is unstable or edge is offline and restarted.
so add test for this function.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: tests/e2e/scripts/execute.sh [hunk 1]
```diff
@@ -51,6 +53,16 @@ kubectl create clusterrolebinding system:anonymous --clusterrole=cluster-admin -
...
+#failed cases number
+fail=`grep -e "SUCCESS\!" -e "FAIL\!" /tmp/testcase.log | awk '{print $6}' | sed -r "s/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[mGK]//g" | awk '{sum+=$1} END {print sum}'`
+
+#test Edge Autonomy
+#only when the above are all successful, we need to test Edge Autonomy function.
+if [ "$fail" == "0" ]
+  then
+    bash ${E2E_DIR}/scripts/autonomy.sh
+fi
+
```
Comment (fisherxu) [inline tests/e2e/scripts/execute.sh:65]: It's better to write this whole logic in deployment_test.go, we can exec shell command in go code.

Comment (gy95) [reply tests/e2e/scripts/execute.sh:65]: got it.

File: tests/e2e/deployment/deployment_test.go [hunk 2]
```diff
@@ -194,4 +196,48 @@ var _ = Describe("Application deployment test in E2E scenario", func() {
...
+
+			gopath, err := utils.GetGOPATH()
+			if err != nil {
+				utils.Fatalf("failed to get GOPATH: %v", err)
+				return
+			}
+			utils.Infof("gopath is %s", gopath)
+
+			script := path.Join(gopath, "src/github.com/kubeedge/kubeedge/tests/e2e/scripts/autonomy.sh")
+			utils.Infof("autonomy scripts path is %v", script)
```
Comment (gy95) [inline tests/e2e/deployment/deployment_test.go:223]: failed to locate the correct path of the shell script. and ref the CI result, the log is 
```
Nov  1 02:54:37.594: Info: gopath is /home/runner/go
Nov  1 02:54:37.595: Info: autonomy scripts path is /home/runner/go/src/github.com/kubeedge/kubeedge/tests/e2e/scripts/autonomy.sh
Nov  1 02:54:37.598: Info: autonomy.sh exec output is

Nov  1 02:54:37.598: Fatal: exec autonomy scripts failed: exit status 127
```
but the code path is 
`/home/runner/work/kubeedge/kubeedge/tests/e2e/deployment/deployment_test.go:230`

code is not in the `GOPATH` dir, I think this method it not correct.
It's easy to locate the go source code dir in the script, but it's not easy to locate the shell script path in the go source code directory.
